src/wallet.cpp

```diff
@@ -1387,113 +1387,171 @@ int64_t CWallet::GetNewMint() const
     return nTotal;
 }
 
-bool CWallet::SelectCoinsMinConf(int64_t nTargetValue, unsigned int nSpendTime, int nConfMine, int nConfTheirs, vector<COutput> vCoins, set<pair<const CWalletTx*,unsigned int> >& setCoinsRet, int64_t& nValueRet) const
+// Our comparator
+struct smallestcoinscomp
 {
-    setCoinsRet.clear();
-    nValueRet = 0;
+    bool operator() (const COutput a, const COutput b)
+    {
+        const CWalletTx* acoin = a.tx;
+        const CWalletTx* bcoin = b.tx;
 
-    // List of values less than target
-    pair<int64_t, pair<const CWalletTx*,unsigned int> > coinLowestLarger;
-    coinLowestLarger.first = std::numeric_limits<int64_t>::max();
-    coinLowestLarger.second.first = NULL;
-    vector<pair<int64_t, pair<const CWalletTx*,unsigned int> > > vValue;
-    int64_t nTotalLower = 0;
+        return (acoin->vout[a.i].nValue < bcoin->vout[b.i].nValue);
+    }
+};
 
-    random_shuffle(vCoins.begin(), vCoins.end(), GetRandInt);
+bool CWallet::SelectCoinsMinConf(int64_t nTargetValue, unsigned int nSpendTime, int nConfMine, int nConfTheirs, vector<COutput> vCoins, set<pair<const CWalletTx*,unsigned int> >& setCoinsRet, int64_t& nValueRet, bool smallest) const
+{
+    setCoinsRet.clear();
+    nValueRet = 0;
 
-    for (auto output : vCoins)
+    if (smallest)
     {
-        const CWalletTx *pcoin = output.tx;
+        // For vote and beacon; users want to use lowest input; lets try to find one or more to meet that specification
+        // We will follow normal rules as well
+        sort(vCoins.begin(), vCoins.end(), smallestcoinscomp());
```

Comment:
I'm trying to wrap my head around the difference between the old and the new implementation. Is it not possible to either sort the input or random shuffle, then use the same coin selection implementation for both?

Reply:
off top of my head (will review this again later)

old implementation:

Gets a list of coins sent in by vCoins.
Then it will random shuffle the inputs.
Then verify they are usable one by one until the target amount of coins needed are met (if not continues in the loops)
In the continuing of the loop it will do various bitcoin logic until there is enough inputs to use for the tx.

My implementation:

using smaller bool it will get the list of coins and sort from lowest to largest then use txs from teh smallest to where ever is needed to meet the coins target to be spent. This in turn will always use the lowest coin inputs first

Reply:
Can you do another test run to see if we only have to sort the list and use the old implementation for coin selection? In pseudo code:

```
if(smallest)
    inputs.sort();
else
    inputs.randomize();
```